### DISCUSSION TITLE
How do you change the path or filename of the renderer log?

### DISCUSSION THREAD

**Discussion by petestarkey** [megahertz/electron-log#304] [Q&A] [answered]
I can easily set the path or filename of the main.log using this in main.ts:

```
log.transports.file.resolvePath = (variables) => {
  return path.join(variables.electronDefaultDir + "whatever", variables.fileName);
}
```

But how do I do the same for the renderer process please?

**Comment by megahertz** [comment]
Run the same code in a renderer process. In the current version, it's configured separately

**Comment by petestarkey** [comment]
Thanks @megahertz. But if I run that same code in the renderer, I get this:

`TypeError
Cannot read properties of undefined (reading 'file')`.

Same error if I use `window.log.transports.file.resolvePath`... too.

**Reply by megahertz** [reply]
Could you post the stacktrace?

**Reply by petestarkey** [reply]
Call Stack
 App
  main.0e4c95e59bcbca7f1e68.hot-update.js:71:39
 renderWithHooks
  renderer.dev.js:22796:18
 updateFunctionComponent
  renderer.dev.js:25167:20
 beginWork
  renderer.dev.js:26874:16
 HTMLUnknownElement.callCallback
  renderer.dev.js:11756:14
 Object.invokeGuardedCallbackDev
  renderer.dev.js:11805:16
 invokeGuardedCallback
  renderer.dev.js:11867:31
 beginWork$1
  renderer.dev.js:31770:7
 performUnitOfWork
  renderer.dev.js:30582:12
 workLoopSync
  renderer.dev.js:30513:5

**Reply by megahertz** [reply]
Nothing helpful in the stack. Unfortunately, there's no way to find the actual reason in that case, since it depends on webContents and bundler settings.  As a workaround for such a case, it's better to use logger in the main process only and use ipc to send log messages from renderer. I have plans to fix that in the next major release, but I have no exact dates for that.

**Comment by petestarkey** [comment, marked as answer]
> Nothing helpful in the stack. Unfortunately, there's no way to find the actual reason in that case, since it depends on webContents and bundler settings. As a workaround for such a case, it's better to use logger in the main process only and use ipc to send log messages from renderer. I have plans to fix that in the next major release, but I have no exact dates for that.

Thanks @megahertz. Marking this as the answer. I used IPC to send messages through to main process as you suggest and this works well. Also means you get just one log file which is a bit neater.
